src/Uno.UI.Runtime.Skia.X11/X11ClipboardExtension.cs

```diff
@@ -406,126 +406,152 @@ private bool SwitchTargets(IntPtr requestor, IntPtr target, IntPtr property)
 		return true;
 	}
 
+	// We rely on the main thread to die in order to kill this background thread. This is fine
+	// since the clipboard loop is shared between all windows, so we only need it to terminate after
+	// all windows are closed.
 	[DoesNotReturn]
-	private void ClipboardOwnerLoop()
+	private unsafe void ClipboardOwnerLoop()
 	{
+		var fds = stackalloc X11Helper.Pollfd[1];
+		fds[0].fd = XLib.XConnectionNumber(_x11Window.Display);
+		fds[0].events = X11Helper.POLLIN;
+
 		while (true)
 		{
 			if (this.Log().IsEnabled(LogLevel.Trace))
 			{
 				this.Log().Trace("XCLIP: Clipboard owner loop, waiting for events.");
 			}
 
-			SpinWait.SpinUntil(() =>
-			{
-				using var _ = X11Helper.XLock(_x11Window.Display);
-				return X11Helper.XPending(_x11Window.Display) > 0;
-			});
+			var ret = X11Helper.poll(fds, 1, -1); // infinite waiting
 
-			using (X11Helper.XLock(_x11Window.Display))
+			if (ret < 0)
 			{
-				XLib.XNextEvent(_x11Window.Display, out var event_);
-
-				if (this.Log().IsEnabled(LogLevel.Trace))
+				if (this.Log().IsEnabled(LogLevel.Error))
 				{
-					this.Log().Trace($"XSEL EVENT: {event_.type}");
+					this.Log().Error("Polling for X11 events failed, defaulting to SpinWait");
```

Comment:
If it's a behavior that the user can't really handle by him self, let's put this as a debug message for now.

Reply:
I don't have a reasonable scenario where this can fail (the X11 calls use the same socket we're polling, so it must be readable), so it's an Error because an informed user would be able to make the connection to the high cpu utilization and it indicates that something weird must have happened so the user should report this is a bug.